File: JavaScriptPractice/Leetcode1.js
var twoSum = function(nums, target) 
{
    let map = {};
    for (let i = 0; i < nums.length; i++) {
       let complement = target  - nums[i];
       if (map[complement]!=undefined) 
       {
            return[i,map[complement]]
       }
       else
       {
            map[nums[i]] = i;
       }
    }
    return [];
    
};

File: CppPractice/CppLeetcode/Leetcode1.cpp
#include <map>
#include <vector>

using namespace std;

class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) 
    {
        vector<int> ans;
        map<int,int> theMap;
        for (int i = 0; i < nums.size(); i++)
        {
            int complement = target - nums[i];
            if (theMap.find(complement)!=theMap.end())
            {
                return {i,theMap[complement]};
            }
            else
            {

                theMap[nums[i]] = i ;

            }
            
            
        }

        vector<int> empty;
        return empty;
        
        
    }
};